openquake/hazardlib/gsim/dowrickrhoades_2005.py

```diff
@@ -0,0 +1,318 @@
+
+        return mean, stddevs
+
+    def _compute_mean(self, C, mag, rrup, hypo_depth, delta_R, delta_S,
+                      delta_V, delta_I, vs30):
+
+        """
+        Compute MMI Intensity Value as per Equation in Table 5 and
+        Table 7 pag 198.
+        """
+        # mean is calculated for all the 4 classes using the same equation.
+        # For DowrickRhoades2005SSlab, the coefficients which don't appear in
+        # Model 3 equationare assigned to zero
+
+        mean = (C['A1'] + (C['A2'] + C['A2R'] * delta_R + C['A2V'] * delta_V) *
+                mag + (C['A3'] + C['A3S'] * delta_S + C['A3V'] * delta_V) *
+                np.log10(np.power((rrup**3 + C['d']**3), 1.0 / 3.0)) +
+                C['A4'] * hypo_depth + C['A5'] * delta_I)
+
+        # Get S site class term
+        S = self._get_site_class(vs30, mean)
+
+        # Add S amplification term to mean value
+        mean = mean + S
+
+        return mean
+
+    def _get_stddevs(self, C, stddev_types, num_sites):
+
+        """
+        Return total standard deviation as described in paragraph 5.2 pag 200.
+        """
+
+        # interevent stddev
+        sigma_inter = C['tau'] + np.zeros(num_sites)
+        # intraevent std
+        sigma_intra = C['sigma'] + np.zeros(num_sites)
+        std = []
+
+        for stddev_type in stddev_types:
+            if stddev_type == const.StdDev.TOTAL:
+                # equation in section 5.2 page 200
+                std += [np.sqrt(sigma_intra**2 + sigma_inter**2)]
+            elif stddev_type == const.StdDev.INTRA_EVENT:
+                std.append(sigma_intra)
+            elif stddev_type == const.StdDev.INTER_EVENT:
+                std.append(sigma_inter)
+
+        return std
+
+    def _get_site_class(self, vs30, mmi_mean):
+        """
+        Return site class flag for:
+        Class E - Very Soft Soil        vs30 < 180
+        Class D - Deep or Soft Soil     vs30 >= 180 and vs30 <= 360
+        Class C - Shallow Soil          vs30 > 360 and vs30 <= 760
+        Class B - Rock                  vs30 > 760 and vs30 <= 1500
+        Class A - Strong Rock           vs30 >= 180 and vs30 <= 360
+        The S site class is equal to
+            S = c1 if MMI <= 7
+            S = c1 - d *(MMI - 7.0) if 7<MMI<9.5
+            S = c2 if MMI >= 9.5
+        """
+
+        if vs30[0] < 180:
+            c1 = 1.0
+            c2 = -0.25
+            d = 0.5
+        elif vs30[0] >= 180 and vs30[0] <= 360:
+            c1 = 0.5
+            c2 = -0.125
+            d = 0.25
+        elif vs30[0] > 360 and vs30[0] <= 760:
+            c1 = 0.
+            c2 = 0.
+            d = 0.
+        elif vs30[0] > 760 and vs30[0] <= 1500:
+            c1 = -0.5
+            c2 = 0.125
+            d = -0.25
+        elif vs30[0] > 1500:
+            c1 = -1.0
+            c2 = 0.25
+            d = -0.5
+
+        S = np.zeros_like(vs30)
+
+        for i in range(vs30.size):
```

Comment:
All the changes requested have been applied.

![image](https://cloud.githubusercontent.com/assets/12708485/11769374/8e887e78-a24a-11e5-8fe4-d57bc308bdb2.png)